core/src/main/scala/chisel3/TestMarker.scala

```diff
@@ -0,0 +1,137 @@
+    *
+    * @example
+    * The following creates a module marked as a formal test:
+    *
+    * {{{
+    * class TestHarness extends RawModule {
+    *   FormalTest(this)
+    * }
+    * }}}
+    *
+    * Additional parameters may be passed to the test, which other tools may use
+    * to control how the test is interpreted or executed:
+    *
+    * {{{
+    * class TestHarness extends RawModule {
+    *   FormalTest(
+    *     this,
+    *     MapTestParam(Map(
+    *       "a" -> IntTestParam(9001),
+    *       "b" -> DoubleTestParam(13.37),
+    *       "c" -> StringTestParam("hello"),
+    *       "d" -> ArrayTestParam(Seq(
+    *         IntTestParam(9001),
+    *         StringTestParam("hello")
+    *       )),
+    *       "e" -> MapTestParam(Map(
+    *         "x" -> IntTestParam(9001),
+    *         "y" -> StringTestParam("hello"),
+    *       ))
+    *     ))
+    *   )
+    * }
+    * }}}
+    */
+  def apply(
+    module: BaseModule,
+    params: MapTestParam = MapTestParam(Map.empty),
+    name:   String = ""
+  )(implicit sourceInfo: SourceInfo): Unit = {
+    val proposedName = if (name != "") {
+      name
+    } else {
+      module._proposedName
+    }
+    val sanitizedName = Builder.globalNamespace.name(proposedName)
+    Builder.components += DefTestMarker("formal", sanitizedName, module, params, sourceInfo)
+  }
+}
+
+object SimulationTest {
+
+  /** Mark a module as a simulation test.
+    *
+    * Other tools can use this information to, for example, collect all modules
+    * marked as simulation tests and run them in a simulator. This is
+    * particularly useful in combination with the `UnitTest` trait.
+    *
+    * @param module The module to be marked.
+    * @param params Optional user-defined test parameters.
+    * @param name Optional name for the test. Uses the module name by default.
+    *
+    * @example
+    * The following creates a module marked as a simulation test:
+    *
+    * {{{
+    * class TestHarness extends RawModule {
+    *   SimulationTest(this)
+    * }
+    * }}}
+    *
+    * Additional parameters may be passed to the test, which other tools may use
+    * to control how the test is interpreted or executed:
+    *
+    * {{{
+    * class TestHarness extends RawModule {
+    *   SimulationTest(
+    *     this,
+    *     MapTestParam(Map(
+    *       "a" -> IntTestParam(9001),
+    *       "b" -> DoubleTestParam(13.37),
+    *       "c" -> StringTestParam("hello"),
+    *       "d" -> ArrayTestParam(Seq(
+    *         IntTestParam(9001),
+    *         StringTestParam("hello")
+    *       )),
+    *       "e" -> MapTestParam(Map(
+    *         "x" -> IntTestParam(9001),
+    *         "y" -> StringTestParam("hello"),
+    *       ))
+    *     ))
+    *   )
+    * }
+    * }}}
+    */
+  def apply(
+    module: BaseModule,
+    params: MapTestParam = MapTestParam(Map.empty),
+    name:   String = ""
```

Comment:
My knee jerk reaction here is to use `Option[String]`. However, that then means you can have `Some("")` which doesn't make sense. This is fine.